src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/service/UserGroupService.kt

```diff
@@ -3,27 +3,61 @@ package uk.gov.justice.digital.hmpps.externalusersapi.service
 import com.microsoft.applicationinsights.TelemetryClient
 import org.hibernate.Hibernate
 import org.slf4j.LoggerFactory
+import org.springframework.data.repository.findByIdOrNull
 import org.springframework.security.core.GrantedAuthority
 import org.springframework.stereotype.Service
 import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.externalusersapi.config.AuthenticationFacade
 import uk.gov.justice.digital.hmpps.externalusersapi.jpa.repository.GroupRepository
 import uk.gov.justice.digital.hmpps.externalusersapi.jpa.repository.UserRepository
 import uk.gov.justice.digital.hmpps.externalusersapi.model.Group
 import uk.gov.justice.digital.hmpps.externalusersapi.model.User
 import uk.gov.justice.digital.hmpps.externalusersapi.security.MaintainUserCheck.Companion.canMaintainUsers
+import java.util.UUID
 
 @Service
 @Transactional(readOnly = true)
 class UserGroupService(
   private val userRepository: UserRepository,
   private val groupRepository: GroupRepository,
-  private val telemetryClient: TelemetryClient
+  private val telemetryClient: TelemetryClient,
+  private val authenticationFacade: AuthenticationFacade
 ) {
 
   companion object {
     private val log = LoggerFactory.getLogger(this::class.java)
   }
 
+  @Transactional
+  @Throws(UserGroupException::class, UserGroupManagerException::class, UserLastGroupException::class)
+  fun removeGroupByUserId(
+    userId: String,
+    groupCode: String
+  ) {
+    userRepository.findByIdOrNull(UUID.fromString(userId))?.let { user ->
+      val groupFormatted = formatGroup(groupCode)
+      if (user.groups.map { it.groupCode }.none { it == groupFormatted }
+      ) {
+        throw UserGroupException("group", "missing")
+      }
```

Comment:
I'm surprised this isn't a not found exception - but at least it returns a "missing"

Reply:
See comment above - new ticket to be raised